hive/operation.go

```diff
@@ -67,10 +67,25 @@ func (op *Operation) FetchResults(ctx context.Context, schema *TableSchema) (*Re
 	if err != nil {
 		return nil, err
 	}
+	cnt := 0
+	for resp.Results == nil {
```

Comment:
I would like to understand the refetch loop. For long running SQL queries the Impala server may itself cancel the query for various reasons. If we execute the fetch again, we are kind of duplicating that scenario. In my opinion, a nil check on `resp.Results` should suffice to fix the panic due to null pointer dereference.

Reply:
I don't think so. `resp.Results` being nil doesn't imply Impala server has cancel the query, maybe the server is busing now. Go-impala needs take loop to fetch data until context timeout.Otherwise, the caller doesn't know if the query(or insert) is executed successful, the call-result information is important to caller.

Reply:
on second thought I don't understand refetch loop too. my understanding (from reading server code) was — server is blocking until result is ready.